### PR TITLE
Metadata columns

### LINKED ISSUE #100: Implement classes for metadata columns
Write an abstraction for columns:
* columns can be indexed
* columns can be not indexed
In both cases, we need a method to filter the stored values by a specific value: `Roaring Column::filter<v>(v value)`.

We need columns that can store string, pango_lineages and dates.

Also implement the DateBetween filter for date columns.

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #100 

This is still missing a pango lineage column, but that can easily be implemented later and we still lack a proper type definition for pango lineages.

### CODE REVIEW

File: include/silo/storage/raw_base_column.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace silo::storage {
+
+template <typename T>
+class RawBaseColumn {
+  private:
+   std::string column_name;
+   std::vector<T> values;
+
+  public:
+   RawBaseColumn(std::string column_name, std::vector<T> values);
```
Comment (JonasKellerer) [inline include/silo/storage/raw_base_column.h:19]: Same here with the const ref.

File: src/silo/storage/date_column.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+   return final_bitmap;
+}
+
+SortedDateColumn::SortedDateColumn(
+   std::string column_name,
+   std::vector<std::chrono::year_month_day> values
+)
+    : column_name(std::move(column_name)),
+      values(std::move(values)) {}
```
Comment (JonasKellerer) [inline src/silo/storage/date_column.cpp:35]: Same here, const ref?

File: src/silo/storage/dictionary.cpp [hunk 3]
```diff
@@ -354,4 +354,11 @@ uint32_t Dictionary::getRegionCount() const {
    return region_count;
 }
 
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
```
Comment (JonasKellerer) [inline src/silo/storage/dictionary.cpp:357]: Not important, but I think there is also //NOLINTBEGIN and //NOLINTEND

**Review by JonasKellerer — APPROVED** [review]
I like it. Only minor changes, mostly concerning pass py value vs pass by const ref.

File: src/silo/storage/dictionary.cpp (hunk 3 above)
Comment (fengelniederhammer) [reply src/silo/storage/dictionary.cpp:357]: This will resolve itself after a rebase on your branch anyway.

**Pull request merged**